templates/commands/render.go

```diff
@@ -322,6 +328,35 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// validateAndDefaultInputs validates and sets the input values provided by the user.
+func validateAndDefaultInputs(flagInputs map[string]string, spec *model.Spec) error {
+	var result *multierror.Error
+
+	for _, i := range spec.Inputs {
+		val, ok := flagInputs[i.Name.Val]
+		if !ok || val == "" {
+			if i.Default.Val != "" {
+				flagInputs[i.Name.Val] = i.Default.Val
+			} else if i.Required.Val {
+				result = multierror.Append(result, fmt.Errorf("%s", i.Name.Val))
+			}
+		}
+	}
+
+	if result != nil {
+		result.ErrorFormat = func(errors []error) string {
+			out := strings.Builder{}
+
+			for _, e := range errors {
+				out.WriteString(fmt.Sprintf("\n  - %s", e))
+			}
+			return out.String()
+		}
+	}
```

Comment:
Elsewhere in `abc`, we just do

```
var merr error

for ... {
  ...
  if err != nil {
    merr = errors.Join(merr,err)
  }
}
return merr
```

That seems to me to be preferable to this multierror scheme because it avoids this 10 lines of ErrorFormat definition.

Reply:
ack